enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,652 @@
+  topological issues, but are still available to install in CRC via
+  the operator hub. Ideally these would not be visible in the default
+  catalog of available operators on CRC. This issue is present prior
+  to this enhancement, and should be addressed via a separate
+  enhancement.
+
+* A new CRC installer. Although it should simplify the SNC tooling
+  used for CRC, this enhancement proposal does not change the basic
+  approach to installing CRC or producing a VM image for the CRC
+  installer. By making the definition of the content and configuration
+  of the components in a single node developer cluster more orthogonal
+  to the tooling that deploys the cluster, this enhancement could
+  enable distinct future enhancements to improve the build and/or
+  install process used by CRC.
+
+* A new OpenShift installer. This enhancement proposal does not add a
+  new OpenShift installer or involve changes, beyond the ability to
+  specify the cluster profile, to the standard OpenShift installer
+  currently used by the SNC tooling to build CRC installation
+  images. Continued evolution of the existing installer, SNC tooling
+  and other CRC components is expected.
+
+* Functional changes for OpenShift. This enhancement proposal does not
+  result in any functional changes to clusters that do not utilize the
+  'single-node-developer' cluster profile.
+
+## Proposal
+
+The 'single-node-developer' cluster profile will be implemented in two
+phases. The first phase, which is targeted for completion for the OCP
+4.7 release, introduces the new cluster profile, and enables some
+simplification of the SNC image build process by utilizing the profile
+to control which components are included in the cluster. The second
+phase, which might begin during the OCP 4.7 development cycle but will
+continue afterwards, leverages the profile to optimize the resource
+utilization of the components included in the cluster, and may further
+simplify the SNC image build process.
+
+### Phase One
+
+In the initial phase, the following annotation will be added to the
+manifest files for all CVO-managed components (operators, deployments,
+etc.) that are to be included in single node developer clusters:
+
+```
+include.release.openshift.io/single-node-developer: "true"
+```
+
+This annotation will need to be added to the manifests for all the
+OpenShift components that are not currently removed by the snc.sh
+script during the CRC build process. The components that are currently
+removed, and thus should not need the annotation added, include:
+
+* any unneeded components running in the openshift-monitoring namespace
+* most components running in the openshift-insights namespace
+* machine-api-operator
+* machine-config-operator (may need to run once, requiring the annotation)
+* openshift-cloud-credential-operator
+* openshift-cluster-storage-operator
+
+Note that OpenShift monitoring functionality is currently disabled by
```

Comment:
we need to be careful that the monitoring operator does not report any issue in this case.

as you do this today with crc, i assume it will not be a problem, but we may want a ci flow that verifies this in implementation phase.